### PR TITLE
[QueryParam] Nullable array with validation

### PR DESCRIPTION
**Pull request by eliecharra**
Fixes FriendsOfSymfony/FOSRestBundle#960

### CODE REVIEW

**Comment by eliecharra** [comment]
I'm fixing tests ...

File: Request/ParamFetcher.php [hunk 1]
```diff
@@ -136,8 +132,14 @@ public function get($name, $strict = null)
         }
 
         if ($config->array) {
+
+            if (($default !== null || !$strict) || $nullable)
+            {
```
Comment (lsmith77) [inline Request/ParamFetcher.php]: curly bracket needs to be on the previous line

**Comment by lsmith77** [comment]
so with this null will then be converted to array() ?

**Comment by eliecharra** [comment]
@lsmith77 Here is a little summary to clarify the situation, let me know if it's ok for you.

Requirements | Default value | Given value | Strict | Nullable | Result
|-------------------|-------------------|------------------|--------|-------------|---------
| `\d+` | `null`              |      `null`       | `false` |  `false` | `Array()`
| `\d+` | `null`              |      `null`       | `true` |  `true` | `Array()`
| `\d+` | `null`              |      `null`       | `false` |  `true` | `Array()`
| `\d+` | `null`              |      `null`       | `true` |  `false` | Query parameter value of 'X' is not an array
| `\d+` | `42`                |      `null`       | `true` |  `true` | `Array(42)`
| `\d+` | `42`                |      `null`       | `true` |  `false` | Query parameter value of 'X' is not an array
| `\d+` | `42`                |      `null`       | `false` |  `false` | `Array(42)`
| `\d+` | `42`                |      `test`       | `false` |  `true` | `Array(42)`
| `\d+` | `42`                |      `test`       | `true` |  `true` | Query parameter X value 'test' violated a constraint
| `\d+` | `test`              |      `null`       | `true` |  `true` | `Array('test')`
| `\d+` | `null`              |      `test`       | `true` |  `false` | Query parameter X value 'test' violated a constraint
| `\d+` | `null`              |      `Array(1)`       | `false` |  `false` | `Array(1)`
| `\d+` | `null`              |      `Array(1)`       | `true` |  `false` | `Array(1)`
| `\d+` | `null`              |      `Array(1)`       | `true` |  `true` | `Array(1)`

**Comment by lsmith77** [comment]
OK thank you .. maybe something to add to the docs.

**Comment by eliecharra** [comment]
@lsmith77 Yeah I think so, maybe we need to add some other tests, not sure we cover all these cases.
It's quite complicated to reverse the behavior from the code there :wink:

**Comment by lsmith77** [comment]
lets have this sit in master for a bit ..

**Comment by lsmith77** [comment]
I just released 1.5.1 without this fix .. but I also now merged it into the 1.5 branch .. still waiting to see if someone reports issues for a few days since I am worried about causing another regression.

**Comment by lsmith77** [comment]
released as https://github.com/FriendsOfSymfony/FOSRestBundle/releases/tag/1.5.2

**Pull request merged**